src/openrct2-ui/UiContext.cpp

```diff
@@ -119,6 +120,20 @@ class UiContext final : public IUiContext
         {
             SDLException::Throw("SDL_Init(SDL_INIT_VIDEO | SDL_INIT_JOYSTICK)");
         }
+        if (gConfigGeneral.RefreshDPIScaling)
+        {
+            float ddpi, hdpi, vdpi;
+            if (!SDL_GetDisplayDPI(0, &ddpi, &hdpi, &vdpi))
+            {
+                // If DPI can be read, divide DPI by regular DPI (96.0f) and round to nearest 0.25
+                gConfigGeneral.WindowScale = std::round(ddpi / 96.0f * 4.0) / 4.0;
```

Comment:
```suggestion
                constexpr auto regularDPI = 96.0f;
                gConfigGeneral.WindowScale = std::round((ddpi / regularDPI) * 4.0) / 4.0;
```

That also makes the comment redundant. (The general rule is to try and make the code speak, if that is not feasible, a comment is fine)